src/network/model/socket.cc

```diff
@@ -25,6 +25,7 @@
 #include "node.h"
 #include "socket.h"
 #include "socket-factory.h"
+#include "../../internet/model/tcp-socket-factory-impl.h"
```

Comment:
Not sure why `#include ns3/tcp-socket-factory-impl.h` does not work here. Could be that this file is linked before tcp-socke-factory.

Reply:
I think so... and this is the only way for now to include tcp socket factory